geowebcache/web/src/main/resources/log4j2.xml

```diff
@@ -6,30 +6,41 @@
     </CustomLevels>>
     <Appenders>
         <Console name="Console" target="SYSTEM_OUT">
-            <PatternLayout pattern="%date{dd MMM HH:mm:ss} (%t) %-6level [%logger{2}] - %msg%n"/>
+            <PatternLayout pattern="%date{dd MMM HH:mm:ss} (%t) %-6level [%logger{2}] - %msg%n%throwable"/>
         </Console>
-        <RollingFileAppender name="Logfile">
-            <MaxBackupIndex>3</MaxBackupIndex>
-            <MaxFileSize>10MB</MaxFileSize>
-            <PatternLayout pattern="%date{dd MMM HH:mm:ss} (%t) %-6level [%logger{2}] - %msg%n"/>
-        </RollingFileAppender>
+        <RollingFile name="Logfile">
+            <FileName>geowebcache.log</FileName>
```

Comment:
Where does this file end up being located?

Reply:
Looks like there is no file name in the existing setup (it was there, has been commented out):

```
# Debugging log settings
log4j.rootLogger=info, stdout
# ------------- stdout logging ---------------------
log4j.appender.stdout=org.apache.log4j.ConsoleAppender
log4j.appender.stdout.layout=org.apache.log4j.PatternLayout
# Follow GeoServer's logging pattern
log4j.appender.stdout.layout.ConversionPattern=%d{dd MMM HH:mm:ss} (%t) %p [%c{2}] - %m%n

# ------------- file logging ---------------------
log4j.appender.logfile=org.apache.log4j.RollingFileAppender

# Keep three backup files.
log4j.appender.logfile.MaxBackupIndex=3

# Maximum file size is 10 MB
log4j.appender.logfile.MaxFileSize=10MB

# TODO : make platform-independent logging
#log4j.appender.logfile.File=${catalina.home}/webapps/logs/geowebcache.log
#log4j.appender.logfile.File=/tmp/geowebcache.log

log4j.appender.logfile.layout=org.apache.log4j.PatternLayout

log4j.appender.logfile.layout.ConversionPattern=%d{dd MMM HH:mm:ss} (%t) %p [%c{2}] - %m%n

log4j.category.org.geowebcache.seed=info
log4j.category.org.geowebcache.diskquota=info
```

The above is ``/geowebcache/web/target/geowebcache/WEB-INF/classes/log4j.properties``

Reply:
I did download a geowebcache.war from source forge and deploy in tomcat, creates in in the catalina.home (not very helpful)

Reply:
Maybe put it in the GWC data dir instead? However, this means reconfiguring the logging on the fly, once you have determined where the data dir is. (and keep only the console logger going up to that point).

Reply:
I turned off the file logigng, matching previous functionality (that only logged if you provided a file name.
I did include an example file, that can be switched to using a system property.